java/trace/src/main/scala/org/typelevel/otel4s/java/trace/TracerImpl.scala

```diff
@@ -59,4 +62,15 @@ private[java] class TracerImpl[F[_]: Sync](
 
   def noopScope[A](fa: F[A]): F[A] =
     scope.noopScope(fa)
+
+  def joinOrContinue[A, C: TextMapGetter](carrier: C)(fa: F[A]): F[A] = {
+    val context = propagators.textMapPropagator.extract(Vault.empty, carrier)
```

Comment:
Should we pass the currently available `Vault` from the `Local[F, Vault]`?
The propagation works either way.

Reply:
That's an excellent question.  Is this a stateful call, or just a function of the carrier?

Not a ton of precedent, but this example in the Java implementation uses the equivalent of the local vault: https://github.com/open-telemetry/opentelemetry-java/blob/2babc699242f3f46e10dd8ede293e698e414d2e5/context/src/main/java/io/opentelemetry/context/propagation/ContextPropagators.java#L51-L59

Reply:
The `Vault` is used as a temporary storage of the **extracted** information. Then we use that vault to rebuild the span context.

Under the hood, we extract `JContext` from the `Vault`. When we pass `Vault.empty`, we automatically fall back to `JContext.root`. 

If we unwrap the call stacks, we have the following flow:
```
toJContext(Vault.empty) (basically, it's JContext.root) -> jPropagator.extract -> fromJContext
```

OpenTelemetry implementation claims the following:
[The returned Context should contain the extracted data, if any, merged with the data from the passed-in Context.](https://github.com/open-telemetry/opentelemetry-java/blob/2babc699242f3f46e10dd8ede293e698e414d2e5/context/src/main/java/io/opentelemetry/context/propagation/TextMapPropagator.java#L115-L119)

_____

In our implementation, if the context has not been extracted from the carrier, we remain in the parent span:

https://github.com/typelevel/otel4s/pull/149/files#diff-fcf3e53e26490232e07d9bdb9f0056edeb09230b4396ce48ffdeb5744dea9e74R69-R73

So, the parent details are not lost.
_____

It's worth mentioning that `JContext` may carry extra information associated with different keys. But this behavior is discouraged 
https://github.com/open-telemetry/opentelemetry-java/blob/2babc699242f3f46e10dd8ede293e698e414d2e5/context/src/main/java/io/opentelemetry/context/Context.java#L66-L69

If I get it right, in our implementation, there is no way to modify the underlying `JContext` and attach extra information. Because we encapsulate all operations over underlying `JContext` and we have complete control over it.
_____

According to the information above, I guess it's OK to pass an empty vault. 

On the other hand, replicating OpenTelemetry behavior is not a big deal and wouldn't hurt us.

Reply:
> On the other hand, replicating OpenTelemetry behavior is not a big deal and wouldn't hurt us.

Where the spec is unclear, replicating existing behavior is a good default, absent a great reason to diverge.